File: user-api/go-sdk/pkg/models/shared/loginparameters.go
package shared

type LoginParametersOauthResponseTypeEnum string

const (
	LoginParametersOauthResponseTypeEnumCode  LoginParametersOauthResponseTypeEnum = "code"
	LoginParametersOauthResponseTypeEnumToken LoginParametersOauthResponseTypeEnum = "token"
)

type LoginParameters struct {
	CognitoIdentityPoolID   *string                               `json:"cognito_identity_pool_id,omitempty"`
	CognitoOauthDomain      *string                               `json:"cognito_oauth_domain,omitempty"`
	CognitoOauthScopes      []string                              `json:"cognito_oauth_scopes,omitempty"`
	CognitoRegion           *string                               `json:"cognito_region,omitempty"`
	CognitoUserPoolClientID *string                               `json:"cognito_user_pool_client_id,omitempty"`
	CognitoUserPoolID       *string                               `json:"cognito_user_pool_id,omitempty"`
	OauthResponseType       *LoginParametersOauthResponseTypeEnum `json:"oauth_response_type,omitempty"`
	OrganizationID          *string                               `json:"organization_id,omitempty"`
	OrganizationName        *string                               `json:"organization_name,omitempty"`
	OrganizationType        *string                               `json:"organization_type,omitempty"`
}

File: user-api/ts-sdk/src/sdk/models/shared/loginparameters.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";


export enum LoginParametersOauthResponseTypeEnum {
    Code = "code",
    Token = "token"
}


export class LoginParameters extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=cognito_identity_pool_id" })
  cognitoIdentityPoolId?: string;

  @SpeakeasyMetadata({ data: "json, name=cognito_oauth_domain" })
  cognitoOauthDomain?: string;

  @SpeakeasyMetadata({ data: "json, name=cognito_oauth_scopes" })
  cognitoOauthScopes?: string[];

  @SpeakeasyMetadata({ data: "json, name=cognito_region" })
  cognitoRegion?: string;

  @SpeakeasyMetadata({ data: "json, name=cognito_user_pool_client_id" })
  cognitoUserPoolClientId?: string;

  @SpeakeasyMetadata({ data: "json, name=cognito_user_pool_id" })
  cognitoUserPoolId?: string;

  @SpeakeasyMetadata({ data: "json, name=oauth_response_type" })
  oauthResponseType?: LoginParametersOauthResponseTypeEnum;

  @SpeakeasyMetadata({ data: "json, name=organization_id" })
  organizationId?: string;

  @SpeakeasyMetadata({ data: "json, name=organization_name" })
  organizationName?: string;

  @SpeakeasyMetadata({ data: "json, name=organization_type" })
  organizationType?: string;
}
